etna/models/sarimax.py

```diff
@@ -0,0 +1,426 @@
+            if `trend='t'` the trend is equal to 1, 2, ..., nobs. Typically is only
+            set when the model created by extending a previous dataset.
+        use_exact_diffuse : bool, optional
+            Whether or not to use exact diffuse initialization for non-stationary
+            states. Default is False (in which case approximate diffuse
+            initialization is used).
+        dates : array_like of datetime, optional
+            If no index is given by `endog` or `exog`, an array-like object of
+            datetime objects can be provided.
+        freq : str, optional
+            If no index is given by `endog` or `exog`, the frequency of the
+            time-series may be specified here as a Pandas offset or offset string.
+        missing : str
+            Available options are 'none', 'drop', and 'raise'. If 'none', no nan
+            checking is done. If 'drop', any observations with nans are dropped.
+            If 'raise', an error is raised. Default is 'none'.
+        """
+        self.order = order
+        self.seasonal_order = seasonal_order
+        self.trend = trend
+        self.measurement_error = measurement_error
+        self.time_varying_regression = time_varying_regression
+        self.mle_regression = mle_regression
+        self.simple_differencing = simple_differencing
+        self.enforce_stationarity = enforce_stationarity
+        self.enforce_invertibility = enforce_invertibility
+        self.hamilton_representation = hamilton_representation
+        self.concentrate_scale = concentrate_scale
+        self.trend_offset = trend_offset
+        self.use_exact_diffuse = use_exact_diffuse
+        self.dates = dates
+        self.freq = freq
+        self.missing = missing
+        self.validate_specification = validate_specification
+        self.kwargs = kwargs
+        self._model = None
+        self._result = None
+
+    def fit(self, df: pd.DataFrame) -> "_SARIMAXModel":
+        """
+        Fits a SARIMAX model.
+
+        Parameters
+        ----------
+        df: pd.DataFrame
+            Features dataframe
+
+        Returns
+        -------
+        self: SARIMAX
+            fitted model
+        """
+        categorical_cols = df.select_dtypes(include=["category"]).columns.tolist()
+        try:
+            df.loc[:, categorical_cols] = df[categorical_cols].astype(int)
+        except ValueError:
+            raise ValueError(
+                f"Categorical columns {categorical_cols} can not been converted to int.\n "
+                "Try to encode this columns manually."
+            )
+
+        self._check_df(df)
+
+        targets = df["target"]
+        targets.index = df["timestamp"]
+
+        exog_train = self._select_regressors(df)
+        regressor_columns = None
+        if not isinstance(exog_train, type(None)):
+            regressor_columns = [col for col in df.columns if col.startswith("regressor")]
```

Comment:
isn't it just a 
```suggestion
            regressor_columns = exog_train.columns
```
?

Reply:
Сори, я просто на if отвлекся и забыл поправить)